### PR TITLE
Filters on small screens js

### PR DESCRIPTION
**Pull request by willmcb**
## Context
This PR adds the filter in a progressive way to devices with small
screens.

**When JS enabled:**

- on large screens the filter is present and the 'hide/show filter' button is not present.

- on small screens the the filter collapses to a floating box on the side and can be
hidden and shown.

**When JS is not enabled:**

- on large screens the filter is present on the left.

- on small screens the filter is moved above the application cards.

## Changes proposed in this pull request

**Before:**
https://recordit.co/ZSXLmxc94b

**After:**
https://recordit.co/ZSXLmxc94b

Without JS:
https://recordit.co/UFN6vpfUqE

## Guidance to review

- checkout this branch - resize the screen. Do things look right to you (ignoring the slightly weird header on small screens there's already a ticket for that).
- As has been discussed there there are no tests for this functionality given that we can't run a headless browser that resizes the screen. 

## Link to Trello card

https://trello.com/c/uIR5I9M4/2122-title-filters-not-working-at-all-on-small-screens

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/frontend/styles/_filter.scss [hunk 1]
```diff
@@ -31,8 +31,7 @@
 }
 
 .app-filter {
-  display: none;
-  @include govuk-media-query($from: large) {
-    display: block;
+  @include govuk-media-query($from: tablet) {
+    margin-bottom: govuk-spacing(3);
```
[the remark this answers is not in the document]
Comment (willmcb) [reply app/frontend/styles/_filter.scss:50]: Hey Duncan - I had a conversation with @adamsilver about this. This is the default behaviour of the Gov.Uk front end. He said that perhaps in a further PR we could do this, but on top is fine for this PR.

[the remark this answers is not in the document]
Comment (duncanjbrown) [reply app/frontend/styles/_filter.scss:50]: Great :+1:

File: package.json [hunk 2]
```diff
@@ -2,10 +2,11 @@
   "name": "apply-for-postgraduate-teacher-training",
   "private": true,
   "dependencies": {
-    "@ministryofjustice/frontend": "0.0.17-alpha",
+    "@ministryofjustice/frontend": "^0.0.19-alpha",
     "@rails/webpacker": "^5.1.1",
     "accessible-autocomplete": "^2.0.2",
-    "govuk-frontend": "^3.7.0"
+    "govuk-frontend": "^3.7.0",
+    "jquery": "^3.5.1"
```
[the remark this answers is not in the document]
Comment (willmcb) [reply package.json:9]: That is a good point - I think it probably would make sense. Something to be aware of though is that jQuery 1.x and 2.x are end of life and unsupported: https://github.com/jquery/jquery.com/issues/162 

Is this something we are okay with?

File: yarn.lock [hunk 3]
```diff
@@ -1069,12 +1069,12 @@
...
-  version "0.0.17-alpha"
-  resolved "https://registry.yarnpkg.com/@ministryofjustice/frontend/-/frontend-0.0.17-alpha.tgz#a5e2e2d2dce58102e9245153bd6d4992e7977c5b"
-  integrity sha512-UOwqFA24kd8xJY6XrWxaQxM2xzr8BHdwb5L82xPk3rG7hgEnWRppk7vt9eszqSt3Hwdfj63pD8aaMTMsi89P4Q==
+"@ministryofjustice/frontend@^0.0.19-alpha":
+  version "0.0.19-alpha"
+  resolved "https://registry.yarnpkg.com/@ministryofjustice/frontend/-/frontend-0.0.19-alpha.tgz#4f00cb02cafef3f8cbfa9d4665885fa766309cdf"
+  integrity sha512-iNNJSIr+poRiBWxyoi4pxs2qcYRfFzVO9yh6ALPUeKDj9pCm1lWYReWhr02/jLgLEJ3nKdJDxIbGv0c/RxSR/w==
   dependencies:
-    moment "^2.22.2"
+    moment "^2.26.0"
```
Comment (tvararu) [inline yarn.lock:1077]: I just noticed this depends on moment... I get the feeling our JS bundle might be a lot bigger than it needs to be. (70kb! https://bundlephobia.com/result?p=moment@2.26.0)

File: app/frontend/styles/_filter.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamsilver) [reply app/frontend/styles/_filter.scss:50]: The MOJ filter layout works by having the filters on the left. It's a very specific grid layout.

@duncanjbrown what you said about the intention is right. But what @willmcb said in terms of doing it later is also fine with me and keeps things chugging along.

**Comment by adamsilver** [comment]
image

Should be this:

image

Give the paragraph a margin-top of 3

Give the filter button a margin bottom of 4

**Pull request merged**